application/src/main/java/uk/nhs/hee/tis/revalidation/service/RecommendationServiceImpl.java

```diff
@@ -254,13 +257,30 @@ public boolean submitRecommendation(String recommendationId, String gmcNumber,
    * @return The last submitted Recommendation or empty
    */
   public TraineeRecommendationRecordDto getLatestRecommendation(String gmcId) {
+    final DoctorsForDB doctorForDB;
+    final var optionalDoctorsForDB = doctorsForDBRepository.findById(gmcId);
+
+    if (optionalDoctorsForDB.isPresent()) {
+      doctorForDB = optionalDoctorsForDB.get();
+    } else {
+      throw new RecommendationException(format(DOCTOR_NOT_FOUND_MESSAGE, gmcId));
+    }
+
     log.info("Fetching latest recommendation info for GmcId: {}", gmcId);
     Optional<Recommendation> optionalRecommendation = recommendationRepository
         .findFirstByGmcNumberOrderByActualSubmissionDateDesc(gmcId);
 
     if (optionalRecommendation.isPresent()) {
-      final var recommendation = optionalRecommendation.get();
+      final Recommendation recommendation = optionalRecommendation.get();
+
+      final boolean isPastCompletedRecommendation = checkIfPastCompletedRecommendation(
+          recommendation,
+          doctorForDB
+      );
 
+      if(isPastCompletedRecommendation){
```

Comment:
By inverting this condition, i.e. `!isPastCompletedRecommendation`, with `buildTraineeRecommendationRecordDto` inside, you can get rid of the call to `new TraineeRecommendationRecordDto();` inside the containing `if()` and therefore  a code branch.